src/sqlparser/src/ast/statement.rs

```diff
@@ -357,11 +361,28 @@ pub(super) fn fmt_create_items(
     columns: &[ColumnDef],
     constraints: &[TableConstraint],
     watermarks: &[SourceWatermark],
+    wildcard_idx: Option<usize>,
 ) -> std::result::Result<String, fmt::Error> {
     let mut items = String::new();
-    let has_items = !columns.is_empty() || !constraints.is_empty() || !watermarks.is_empty();
+    let has_items = !columns.is_empty()
+        || !constraints.is_empty()
+        || !watermarks.is_empty()
+        || wildcard_idx.is_some();
```

Comment:
do we store the `*`'s index in catalog?
Is it true that the columns on the left of `*` means defined schema and on the right means generated columns.

Reply:
It will be in the sql string in catalog

> Is it true that the columns on the left of * means defined schema and on the right means generated columns
No,  * means all columns from external source